### PR TITLE
Vet clinic database: query and update animals table

### PR DESCRIPTION
**Pull request by nilab-khwaja**
Add a column species of type string to animals table. 
Insert  data.
Inside a transaction update the animals table.
Inside a transaction delete all records in the animals table.
Inside a transaction:

- Delete all animals born after Jan 1st, 2022.
- Create a savepoint for the transaction.
- Update all animals' weight to be their weight multiplied by -1.
- Rollback to the savepoint
- Update all animals' weights that are negative to be their weight multiplied by -1.
- Commit transaction
Write queries to answer the following questions:

- How many animals are there?
- How many animals have never tried to escape?
- What is the average weight of animals?
- Who escapes the most, neutered or not neutered animals?
- What is the minimum and maximum weight of each type of animal?
- What is the average number of escape attempts per animal type of those born between 1990 and 2000?   
-

### CODE REVIEW

File: queries.sql [hunk 1]
```diff
@@ -7,4 +7,26 @@ SELECT name, escape_attempts FROM animals WHERE weight_kg > 10.5;
 SELECT * FROM animals WHERE neutered = true;
 SELECT * FROM animals WHERE name != 'Gabumon';
 SELECT * FROM animals WHERE weight_kg BETWEEN  10.5 AND 17.3;
+UPDATE animals SET species = 'unspecified';
```
Comment (iambenkis) [inline queries.sql]: - [ ] They update function should be made inside the transaction that start with `BEGIN` that has a `COMMIT` and the `ROLLBACK` to come back 👍 , Kindly ensure this update fucttion follow these steps 👍 

```sql
BEGIN; -- start transaction

   UPDATE animals
   SET species = 'unspecified'; -- make change

   SELECT species from animals; -- verify that change was made
   ROLLBACK;

   SELECT species from animals; -- verify that change was undone
```

File: queries.sql [hunk 2]
```diff
@@ -7,4 +7,26 @@ SELECT name, escape_attempts FROM animals WHERE weight_kg > 10.5;
...
+BEGIN ;
+UPDATE animals SET species = 'digimon' WHERE name LIKE '%mon';
+UPDATE animals SET species = 'pokimon' WHERE species IS NULL;
+COMMIT ;
```
Comment (iambenkis) [inline queries.sql]: - [ ] Good job update the animals table, 👍 however you should verify that the change has been made before the commit and after the commit 👍 ,Kindly check that 👀 

```sql
BEGIN;
   UPDATE animals SET species = 'digimon' WHERE name LIKE '%mon'; 
   UPDATE animals SET species = 'pokemon' WHERE species IS NULL; 

   SELECT species from animals; -- verify that change was made
   COMMIT;
   SELECT species from animals; -- verify that change persists after commit
```

File: queries.sql [hunk 3]
```diff
@@ -7,4 +7,26 @@ SELECT name, escape_attempts FROM animals WHERE weight_kg > 10.5;
...
+BEGIN;
+DELETE FROM animals;
+ROLLBACK;
```
Comment (iambenkis) [inline queries.sql]: - [ ] Kindly check the differene between after `ROLLBACK` state and before the `ROLLBACK` 👍 to ensure that the change have been made 😄 

```sql
 BEGIN;

   DELETE FROM animals WHERE ;
   SELECT COUNT(*) FROM ANIMALS;

   ROLLBACK;
   SELECT COUNT(*) FROM ANIMALS;
```

File: data.sql [hunk 4]
```diff
@@ -1 +1,8 @@
-insert into animals(name, date_of_birth, escape_attempts, neutered, weight_kg) VALUES ('Agumon', '2020-02-03', 0, true, 10.23), ('Gambumon', '2018-11-15', 2, true, 8),('Pikachu', '2021-01-07', 1, false, 15.04), ('Devimon', '2017-3-12', 5, true, 11);
\ No newline at end of file
+insert into animals(name, date_of_birth, escape_attempts, neutered, weight_kg) VALUES 
+('Agumon', '2020-02-03', 0, true, 10.23), ('Gambumon', '2018-11-15', 2, true, 8),('Pikachu', '2021-01-07', 1, false, 15.04), ('Devimon', '2017-3-12', 5, true, 11);
+
+INSERT INTO animals (name,date_of_birth, escape_attempts, neutered, weight_kg) VALUES
+ ('Charmander', '2022-2-8', 0, false, 11), ('Plantmon', '2021-11-15', 2, true, 5.7),
+ ('Squirtle', '1993-4-2', 3, false, 12.13), ('Angemon', '2005-6-12', 1,true, 45),
+  ('Boarmon', '2005-6-7',7,true, 20.4), ('Blossom', '1998-10-13', 3, true, 17), 
+  ('Ditto', '2022-5-14', 4, true, 22);
```
Comment (iambenkis) [inline data.sql]: - [ ] Good job adding data provided in the project requirements 👍 , however , I will suggest you to pay a closer attention to the fact that certains weight value has negative value , kindly check the [project requirements](https://github.com/microverseinc/curriculum-databases/blob/main/db-structure/vet_clinic_query_update_tables.md) and correct that 😄

**Review by iambenkis — CHANGES_REQUESTED** [review]
#### Hi @nilab-khwaja  

Good job so far!
There are some issues that you still need to work on to prepare your project for the final evaluation but you are almost there!
You did a great job, keep up the good work 👍🏼

giphy


## Highlights

✔️ Tranction feature added  
✔️ Correct screenshots results

## Suggested changes

- [ ] Kindly fix the broken link attached to the[ issue page](https://github.com/nilab-khwaja/Vet_Clinic_DB.git/issues) link in your README file to make it professional 👍 

_Check the comments under the review._

You can use as many of my suggestions as you want. If there is anything you would like to skip - feel free to do that. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

**Review by thecodechaser — APPROVED** [review]
## STATUS: APPROVED 🟢

## Hi @nilab-khwaja👋, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## To Highlight 🎉
- Good job with transactions. ✔️
- Gitflow is followed. :heavy_check_mark:
- Professional README file. :heavy_check_mark:
- PR has a good title and summary. ✔️



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**